misc/package/Data Files/MWSE/core/mcm/components/categories/Category.lua

```diff
@@ -61,18 +61,35 @@ function Category:new(data)
 	return t
 end
 
+
```

Comment:
How would you feel about splitting up the logic a bit so that the `Category` doesn't need to look inside the `Setting`s?

Maybe something like

```lua
-- in Category.lua
function Category:resetSettings()
	for _, component in ipairs(self.components) do
		local compType = component.componentType
		if compType == "Setting" then	
			component:resetToDefaultSetting()
		elseif compType == "Category" or compType == "Page" then
			component:resetSettings()
		end
	end
end

-- in Setting.lua
function Setting:resetToDefaultSetting()
	if self.variable and self.variable.defaultSetting then	
		self:setVariableValue(self.variable.defaultSetting)
	end
end
```
It might also help to have both methods share the same name. E.g., `Category:resetToDefault` and `Setting:resetToDefault`. This way, the implementation could be further simplified by doing something like 

```lua
function Category:resetToDefault()
	for _, component in ipairs(self.components) do
		component:resetToDefault()
	end
end
```
Although that would also necessitate defining an empty `Component:resetToDefault()` method.

I don't feel particularly strongly about any of these ideas, I mainly just wanted to see what you think of them.

Reply:
That's a good point and it makes the implementation a lot cleaner. I didn't know that categories and settings were the only things that could be nested inside a `Category`, but that's very useful to know for the future.